framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1206,6 +1219,32 @@ public List<AnnotatedTypeMirror> getParameterTypes() {
             return paramTypes;
         }
 
+        /**
+         * Sets the varargs types of this executable type.
+         *
+         * @param superCon for anonymous class
+         */
+        /*package-private*/
+        void setVarargType(@Nullable AnnotatedExecutableType superCon) {
+            if (!isVarArgs()) {
+                return;
+            }
+            List<AnnotatedTypeMirror> parameters = paramTypes;
+            if (superCon != null) {
+                parameters = superCon.getParameterTypes();
+            }
+            varargType = (AnnotatedArrayType) parameters.get(parameters.size() - 1);
+        }
```

Comment:
I think we can make this setter simpler and let the caller take care of the rest. e.g.,
```
void setVarargType(List<AnnotatedTypeMirror> parameters) {
    if (!isVarArgs()) {
        return;
    }
    varargType = (AnnotatedArrayType) parameters.get(parameters.size() - 1);
}
```
or even
```
void setVarargType(AnnotatedTypeMirror varargType) {
    if (!isVarArgs()) {
        return;
    }
    this.varargType = varargType;
}
```

Reply:
or overloading both, depending on your use cases.